Editor/EditorCore/CutTool.cs

```diff
@@ -0,0 +1,1421 @@
+        //Snapping
+        int m_SnapedVertexId = -1;
+        Edge m_SnapedEdge = Edge.Empty;
+
+        bool m_SnapToGeometry;
+        float m_SnappingDistance;
+
+        //Overlay fields
+        GUIContent m_OverlayTitle;
+        const string k_SnapToGeometryPrefKey = "VertexInsertion.snapToGeometry";
+        const string k_SnappingDistancePrefKey = "VertexInsertion.snappingDistance";
+
+        public bool IsALoop
+        {
+            get
+            {
+                if (m_CutPath.Count < 3)
+                    return false;
+                else
+                    return Math.Approx3(m_CutPath[0].position,
+                        m_CutPath[m_CutPath.Count - 1].position);
+            }
+        }
+
+        public int ConnectionsToBordersCount
+        {
+            get
+            {
+                return m_CutPath.Count(data => (data.types & (VertexTypes.AddedOnEdge | VertexTypes.ExistingVertex)) != 0
+                                                    && (data.types & VertexTypes.VertexInShape) == 0);
+            }
+        }
+
+        void OnEnable()
+        {
+            s_HandleColorUseExistingVertex = Handles.selectedColor;
+            s_HandleColorAddVertexOnEdge = Handles.selectedColor;
+
+            m_OverlayTitle = new GUIContent("Cut Tool");
+            m_SnapToGeometry = EditorPrefs.GetBool( k_SnapToGeometryPrefKey, false );
+            m_SnappingDistance = EditorPrefs.GetFloat( k_SnappingDistancePrefKey, 0.1f );
+
+            m_CutCursorTexture = Resources.Load<Texture2D>("Cursors/cutCursor");
+            m_CutAddCursorTexture = Resources.Load<Texture2D>("Cursors/cutCursor-add");
+            m_CutMagnetCursorTexture = Resources.Load<Texture2D>("Cursors/cutCursor-magnet");
+            m_CutAddMagnetCursorTexture = Resources.Load<Texture2D>("Cursors/cutCursor-add-magnet");
+
+            Undo.undoRedoPerformed += UndoRedoPerformed;
+            if(MeshSelection.selectedObjectCount == 1)
+                m_Mesh = MeshSelection.activeMesh;
+        }
+
+        void OnDisable()
+        {
+            Undo.undoRedoPerformed -= UndoRedoPerformed;
+
+            ExecuteCut();
+            Clear();
+        }
+
+        /// <summary>
+        /// Clear all data from the cut tool
+        /// </summary>
+        void Clear()
+        {
+            m_Mesh = null;
+            m_TargetFace = null;
+            m_CurrentFace = null;
+            m_PlacingPoint = false;
+            m_CurrentCutCursor = null;
+            m_CutPath.Clear();
+            m_MeshConnections.Clear();
+        }
+
+        /// <summary>
+        /// Undo/Redo callback: Reset and recompute lines, and update the targeted face if needed
+        /// </summary>
+        private void UndoRedoPerformed()
+        {
+            if(m_CutPath.Count == 0)
+                m_TargetFace = null;
+
+            m_SelectedIndex = -1;
+            m_MeshConnections.Clear();
+
+            EditorApplication.delayCall = () => RebuildCutShape();
```

Comment:
why is this done as a delayed call?

Reply:
I think it was for a purpose that is not valid anymore => I removed it